checkout/application/orderService.go

```diff
@@ -62,11 +62,59 @@ const (
+var (
+	// orderValidationFailCount counts validation failures on orders
+	orderValidationFailCount = stats.Int64("flamingo-commerce/checkout/order_validation_failed", "Count of failures while validating orders", stats.UnitDimensionless)
+
+	// decoratedCartCreationFailCount counts creation errors for decorated carts
+	decoratedCartCreationFailCount = stats.Int64("flamingo-commerce/checkout/decorated_cart_creation_failed", "Count of failures while creating decorated cart", stats.UnitDimensionless)
+
+	// noPaymentSelectionCount counts error for orders without payment selection
+	noPaymentSelectionCount = stats.Int64("flamingo-commerce/checkout/no_payment_selection", "Count of orders without having a selected payment", stats.UnitDimensionless)
+
+	// selectedPaymentGatewayNotFoundCount counts errors if payment gateway for selected payment could not be found
+	selectedPaymentGatewayNotFoundCount = stats.Int64("flamingo-commerce/checkout/selected_payment_gateway_not_found", "The selected payment gateway could not be found", stats.UnitDimensionless)
+
+	// paymentFlowStatusErrorCount counts errors while fetching payment flow status
+	paymentFlowStatusErrorCount = stats.Int64("flamingo-commerce/checkout/payment_flow_status_error", "Count of failures while fetching payment flow status", stats.UnitDimensionless)
+
+	// orderPaymentFromFlowErrorCount counts errors while fetching payment from flow
+	orderPaymentFromFlowErrorCount = stats.Int64("flamingo-commerce/checkout/order_payment_from_flow", "Count of failures while fetching payment from flow", stats.UnitDimensionless)
+
+	// paymentFlowStatusFailedCanceledCount counts orders trying to be placed with payment status either failed or canceled
+	paymentFlowStatusFailedCanceledCount = stats.Int64("flamingo-commerce/checkout/payment_flow_status_failed_canceled", "Count of payments with status failed or canceled", stats.UnitDimensionless)
+
+	// paymentFlowStatusAbortedCount counts orders trying to be placed with payment status aborted
+	paymentFlowStatusAbortedCount = stats.Int64("flamingo-commerce/checkout/payment_flow_status_aborted", "Count of payments with status aborted", stats.UnitDimensionless)
+
+	// placeOrderFailCount counts failed placed orders
+	placeOrderFailCount = stats.Int64("flamingo-commerce/checkout/place_order_failed", "Count of failures while placing orders", stats.UnitDimensionless)
+
+	// placeOrderSuccessCount counts successfully placed orders
+	placeOrderSuccessCount = stats.Int64("flamingo-commerce/checkout/place_order_successful", "Count of successfully placed orders", stats.UnitDimensionless)
+)
 
 func init() {
 	gob.Register(PlaceOrderInfo{})
-	opencensus.View("flamingo-commerce/orderfailed/count", orderFailedStat, view.Count())
+	openCensusViews := map[string]*stats.Int64Measure{
+		"flamingo-commerce/checkout/order_validation_failed":             orderValidationFailCount,
```

Comment:
Can we maybe move all the errors to a `order` namespace something like this so it becomes clearer that all these metrics are happening while placing an order?

Reply:
something like this:
```
flamingo-commerce/checkout/orders/cart_validation_failed
flamingo-commerce/checkout/orders/started // new
flamingo-commerce/checkout/orders/failed
flamingo-commerce/checkout/orders/success
flamingo-commerce/checkout/orders/payment/no_payment_selection
flamingo-commerce/checkout/orders/payment/payment_gateway_not_found
flamingo-commerce/checkout/orders/payment/order_payment_from_flow_error
flamingo-commerce/checkout/orders/payment/flow_status_error
flamingo-commerce/checkout/orders/payment/flow_status_failed_canceled
flamingo-commerce/checkout/orders/payment/flow_status_aborted
```

Reply:
Used your suggestion and added the namespace `orders`